frontend/src/static/js/components/webstatus-overview-page.ts

```diff
@@ -67,7 +70,17 @@ export class OverviewPage extends LitElement {
       task: async ([apiClient, routerLocation]): Promise<
         components['schemas']['FeaturePage']
       > => {
-        return this._fetchFeatures(apiClient, routerLocation);
+        if (this.location.search !== this.currentLocation?.search) {
+          // Reset taskTracker here due to a Task data cache issue.
+          this.taskTracker = {
+            status: TaskStatus.INITIAL,
+            error: null,
+            data: null,
+          };
+          this.currentLocation = this.location;
+          return this._fetchFeatures(apiClient, routerLocation);
+        }
+        return {} as components['schemas']['FeaturePage'];
```

Comment:
We should avoid runtime type assertions. Instead we can return, the existing task tracker data (or empty feature page)
```suggestion
        return this.taskTracker.data ?? {metadata: {total: 0}, data: []};
```